### PR TITLE
Extended HSL/HSLA/HSB Handling with Float Support

### PR DESCRIPTION
**Pull request by danielebarbaro**
This PR introduces extended support for handling **HSL**, **HSLA**, and **HSB** color formats with floating-point values. 

The changes address and resolve [Issue #89](https://github.com/spatie/color/issues/89).

**Changes:**
1. Centralized Regex Patterns:
   - Introduced `HsPatterns` class to centralize and manage regex patterns for validation and extraction.
   - This ensures consistency and maintainability of regex patterns across the library.

2. Enhanced Parsing Logic:
    - Updated `Hsb`, `Hsl`, and `Hsla` classes to use the new `HsPatterns` class for parsing.
    - Modified the `fromString` methods to correctly parse and handle floating-point values in color strings.

3. Validation Improvements:
    - "Refactored" Validate methods (`hsbColorString`, `hslColorString`, and `hslaColorString`) to utilize the new centralized regex patterns in `HsPatterns`.

4. Tests and Dataset Enhancements:
    - Added test cases in Colors to cover scenarios involving floating-point values in HSL/HSLA/HSB strings.
    - Examples include edge cases such as hsla(357, 12.12%, 44.5%, 0.1) and hsl(221, 11.52%, 32.35%) (taken from the issue).
    - All updated test cases pass successfully, and new datasets were added to ensure comprehensive coverage for float handling.

Resolves:
[Issue #89](https://github.com/spatie/color/issues/89)

### CODE REVIEW

File: src/HsPatterns.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Spatie\Color;
+
+class HsPatterns
+{
+    private const HUE = '\d{1,3}';
```
Comment (freekmurze) [inline src/HsPatterns.php:7]: Let's make these `protected`

Comment (freekmurze) [reply src/HsPatterns.php:7]: These also can be regular class properties, not constants.

**Review by freekmurze — CHANGES_REQUESTED** [review]

File: src/HsPatterns.php (hunk 1 above)
Comment (danielebarbaro) [reply src/HsPatterns.php]: I decided to change from `private` to `protected`. 

My initial idea was to create a `ColorPattern` class instead of `HsPattern`, but that would have required a complete refactor of the package—and doing that the day before Christmas would have been a bit naughty. 

Using constants _feels_ cleaner to me, as it allows me to avoid touching `VALIDATION_PATTERNS` and `EXTRACTION_PATTERNS`.

Comment (freekmurze) [reply src/HsPatterns.php]: Let's make these regular properties, they can be static

Comment (freekmurze) [reply src/HsPatterns.php]: Ok, those are good arguments, thanks for explaining. 👍

**Comment by freekmurze** [comment]
Thank you!
